apps/rush-lib/src/cli/actions/AddAction.ts

```diff
@@ -0,0 +1,105 @@
+// See LICENSE in the project root for license information.
+
+import * as os from 'os';
+
+import {
+  CommandLineFlagParameter,
+  CommandLineStringParameter
+} from '@microsoft/ts-command-line';
+
+import { RushConfigurationProject } from '../../api/RushConfigurationProject';
+import { BaseRushAction } from './BaseRushAction';
+import { RushCommandLineParser } from '../RushCommandLineParser';
+import { DependencyIntegrator, SemVerStyle } from '../../logic/DependencyIntegrator';
+
+export class AddAction extends BaseRushAction {
+  private _exactFlag: CommandLineFlagParameter;
+  private _caretFlag: CommandLineFlagParameter;
+  private _devDependencyFlag: CommandLineFlagParameter;
+  private _makeConsistentFlag: CommandLineFlagParameter;
+  private _noInstallFlag: CommandLineFlagParameter;
+  private _packageName: CommandLineStringParameter;
+  private _versionSpecifier: CommandLineStringParameter;
+
+  constructor(parser: RushCommandLineParser) {
+    const documentation: string[] = [
+      'Blah.'
+    ];
+    super({
+      actionName: 'add',
+      summary: 'Adds a dependency to the package.json and runs rush upgrade.',
+      documentation: documentation.join(os.EOL),
+      safeForSimultaneousRushProcesses: false,
+      parser
+    });
+  }
+
+  public onDefineParameters(): void {
+    this._packageName = this.defineStringParameter({
+      parameterLongName: '--package',
+      parameterShortName: '-p',
+      required: true,
+      argumentName: 'PACKAGE_NAME',
+      description: '(Required) The name of the package which should be added as a dependency'
+    });
+    this._versionSpecifier = this.defineStringParameter({
+      parameterLongName: '--version',
+      parameterShortName: '-v',
+      argumentName: 'VERSION_RANGE',
+      description: ''
+    });
+    this._exactFlag = this.defineFlagParameter({
+      parameterLongName: '--exact',
+      description: 'If specified, the version specifier inserted into the'
+        + ' package.json will be a locked, exact version.'
+    });
+    this._caretFlag = this.defineFlagParameter({
+      parameterLongName: '--caret',
+      description: 'If specified, the version specifier inserted into the'
+        + ' package.json will be a prepended with a "caret" specifier ("^").'
+    });
+    this._devDependencyFlag = this.defineFlagParameter({
+      parameterLongName: '--dev',
+      description: 'If specified, the package will be added as a "devDependency"'
+        + ' to the package.json'
+    });
+    this._makeConsistentFlag = this.defineFlagParameter({
+      parameterLongName: '--make-consistent',
+      parameterShortName: '-c',
+      description: 'If specified, other packages with this dependency will have their package.json'
+        + ' files updated to point at the specified depdendency'
+    });
+    this._noInstallFlag = this.defineFlagParameter({
+      parameterLongName: '--no-install',
+      parameterShortName: '-n',
```

Comment:
>-n [](start = 27, length = 2)

Is this the same as npm?

Reply:
Wontfix?

---
In reply to: [220407520](https://github.com/Microsoft/web-build-tools/pull/843#discussion_r220407520) [](ancestors = 220407520)

Reply:
What do you mean. NPM doesn't have an option for --skip-install or --no-install or anything. 

---
In reply to: [220750062](https://github.com/Microsoft/web-build-tools/pull/843#discussion_r220750062) [](ancestors = 220750062,220407520)

Reply:
That's what I was asking.

---
In reply to: [220751842](https://github.com/Microsoft/web-build-tools/pull/843#discussion_r220751842) [](ancestors = 220751842,220750062,220407520)